src/index.ts

```diff
@@ -157,6 +165,45 @@ function generateEachResolverType(
   argDefs.forEach(a => chunks.push(a));
 }
 
+function generateFieldSignature(
+  type: GraphQLObjectType | GraphQLInterfaceType,
+  config: Config,
+  interfaceToImpls: Map<GraphQLInterfaceType, GraphQLObjectType[]>,
+  argDefs: Code[],
+) {
+  // For GraphQLObjectType, don't include fields which are coming from an implemented interface
+  const excludeFields =
+    type instanceof GraphQLObjectType ? type.getInterfaces().flatMap(i => Object.keys(i.getFields())) : [];
+
+  return Object.values(type.getFields())
+    .filter(f => !excludeFields.includes(f.name))
+    .map(f => {
+      const argsName = `${type.name}${upperCaseFirst(f.name)}Args`;
+      const args = f.args.length > 0 ? argsName : "{}";
+      if (f.args.length > 0) {
+        argDefs.push(code`
+              export interface ${argsName} {
+                ${f.args.map(a => {
+                  const maybeOptional = isNullableType(a.type) ? "?" : "";
+                  return code`${a.name}${maybeOptional}: ${mapType(config, interfaceToImpls, a.type)}; `;
+                })}
+              }`);
+      }
+
+      const root = type instanceof GraphQLObjectType ? mapObjectType(config, type) : "T";
+      const result = mapType(config, interfaceToImpls, f.type);
+      return code`${f.name}: Resolver<${root}, ${args}, ${result}>;`;
+    });
+}
+
+function extendInterfaces(type: GraphQLObjectType, root: any) {
```

Comment:
Suggestion can we change the `any` and add a `string` return type?

Reply:
done 👍